iotedgedev/modules.py

```diff
@@ -15,6 +14,53 @@ def __init__(self, envvars, utility, output, dock):
         self.dock = dock
         self.dock.init_registry()
 
+    def create(self, name, template):
+        self.output.header("CREATING MODULE")
+
+        repo = "{0}/{1}".format(self.envvars.CONTAINER_REGISTRY_SERVER, name.lower())
+        cwd = self.envvars.MODULES_PATH
+        if template == "csharp":
```

Comment:
This whole block can be refactored.  There's a lot that can be extracted into common modules.  I also don't like the idea of calling exe_proc from this module.  

Can you create a class that wraps "dotnet"?  And then also change other calls to dotnet, like in dotnetmoduleprocessor.py to use that new dotnetcls.py class you create.

Reply:
Thanks. I have added a new `DotNet` module to wrap .NET Core CLI operations. For Node.js and Python module, I don't think it worthwhile to extract the `yo` and `cookiecutter` operations to separate modules since they will be only used when adding new modules (*i.e.*, in this class).